interview-questions.md

```diff
@@ -8,50 +8,80 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter is how we can make dynamic funcitons. We give the funciton value that can goes inside of parentheses in our arrow function set up. This value we can name as whatever we want but best pracitce is the type of data to be used. This is called the parameter. The argument is the actually data we want to pass in when running the code. This is usually found at the end of the funciton contained in the console.log line of code. 
 
-Researched answer:
+Researched answer: The variables that are used when a funcition is created are know as the parameters. The data or values that are used when the function is called are known as arguments.
+
+answer via:https://byjus.com/gate/difference-between-argument-and-parameter-in-c-and-c-plus-plus/#:~:text=1-,The%20values%20that%20are%20declared%20within%20a%20function%20when%20the,declared%20are%20known%20as%20parameters.&text=These%20are%20used%20in%20function,function%20to%20the%20receiving%20function.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: The .map() method can used 3 different predefined parameters. Those parametes are value, array, and index. For the .map method to work there is only one required parameter which is value the other two are optional.
+
+Researched answer: There are 3 possible designated parameters for the .map() method. The first is value, this parameter is the value the method is processing at different steps in the function. The next is index. This is simply the index of the current value. Last is array, which is a parameter that points to the original array. A combination of all three parameters can be used however only value is required to use .map().
 
-Researched answer:
+answer via: https://stackabuse.com/guide-to-javascripts-map-method/
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: Map is a method that will take in an array and create a new array after iterating over the old. We can take an array of all lower case strings and use the .map() method to change them to all uppercase because of its ability to look at each value and make the change. Filter is a method which can be used to extract personlized values from an array. For example if we have an array of 100 numbers we can use .filter() to set a conditional statement that will return only odd numbers. 
+
+Researched answer: the method .map() returns an array of informatin from the original array. We can use it in the call back function to change or get only the data wanted. The .filter() method is used to aquire a subset of data from the original array based on provided condition. 
```

Comment:
This is some good research. Just remember we love to hear your own interpretation of what you've learned. Keep up the good work!